libraries/GCS_MAVLink/GCS_Common.cpp

```
@@ -2433,6 +2433,17 @@ void GCS_MAVLINK::send_heartbeat() const
         system_status());
 }
 
+MAV_RESULT GCS_MAVLINK::handle_command_do_aux_function(const mavlink_command_long_t &packet)
+{
+    if (packet.param2 > 2) {
+        return MAV_RESULT_DENIED;
+    }
+    const RC_Channel::AUX_FUNC aux_func = (RC_Channel::AUX_FUNC)packet.param1;
+    const RC_Channel::AuxSwitchPos position = (RC_Channel::AuxSwitchPos)packet.param2;
+    rc().do_aux_function(aux_func, position);
```

this should fail if invalid aux fn

---

Done